cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -1816,7 +1817,8 @@ class TestPrimitiveQuantileKernel : public ::testing::Test {
     for (auto interpolation : this->interpolations_) {
       options.interpolation = interpolation;
       ASSERT_OK_AND_ASSIGN(Datum out, Quantile(array, options));
-      ASSERT_OK(out.make_array()->ValidateFull());
+      auto out_array = out.make_array();
+      ValidateOutput(*out_array);
```

Comment:
```suggestion
      ValidateOutput(out);
```

Reply:
And actually: it would be simpler and not too expensive to only make the Datum overload public.